### PR TITLE
Introduce ExtendedNumberInput component

### PR DESCRIPTION
**Pull request by Ladsgroup**
Bug: T275114
Bug: T275113

### CODE REVIEW

File: vue-components/stories/ExtendedNumberInput.stories.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    title: 'Wikibase Components/ExtendedNumberInput',
+};
+
+export function basic( args: { label: string, placeholder: string } ): Component {
+    return {
+        components: { ExtendedNumberInput },
+        data(): unknown {
+            return {
+                placeholder: args.placeholder,
+                label: args.label,
```
Comment (micgro42) [inline vue-components/stories/ExtendedNumberInput.stories.ts:18]: You can drop those two data properties as they are already defined as props. That also takes care of the error messages in the console

**Comment by micgro42** [comment]
The storybook nitpick aside, it looks good! Its validation regex is a tiny bit stricter than the original validation when it comes to spacing, but that's ok, we can always loosen it up if that becomes an issue.

Still needs design review by @SaiSan-WMDE

File: vue-components/stories/ExtendedNumberInput.stories.ts (hunk 1 above)
Comment (Ladsgroup) [reply vue-components/stories/ExtendedNumberInput.stories.ts]: Oh I saw it, wanted to investigate but forgot. Thanks. Removed now.

**Comment by SaiSan-WMDE** [comment]
If this sis the extended number input, how come "e" is validated as an error?

File: vue-components/stories/ExtendedNumberInput.stories.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        props: Object.keys( args ),
+        template: `
+          <div>
+          <p><ExtendedNumberInput
+              :label="label"
+              :placeholder="placeholder"
+              :error="error"
+              v-model="value"
+              @input="() => error = null"
+              @invalid-input="() => error = {type: 'error', message: 'bad value'}"
```
Comment (SaiSan-WMDE) [inline vue-components/stories/ExtendedNumberInput.stories.ts:28]:
```suggestion
              @invalid-input="() => error = {type: 'error', message: 'Please enter a valid number'}"
```

Comment (SaiSan-WMDE) [reply vue-components/stories/ExtendedNumberInput.stories.ts:28]: Just to be a bit more descriptive

Comment (Ladsgroup) [reply vue-components/stories/ExtendedNumberInput.stories.ts]: Accepted and committed the change.

**Review by SaiSan-WMDE — APPROVED** [review]
Looking good! Thanks, Amir 💯

**Comment by Ladsgroup** [comment]
I let Michael to do the final approval and we can merge it \o/

**Review by micgro42 — APPROVED** [review]
Fine from my side 👍

**Pull request merged**